ios/ReactNativeLibsignalClientModule.swift

```diff
@@ -226,8 +226,283 @@ open class InMemorySignalProtocolStoreWithPreKeysList: IdentityKeyStore, PreKeyS
+            let clZkCipher = ClientZkGroupCipher(groupSecretParams: gsp)
+            return try clZkCipher.decryptBlob(blobCiphertext: [UInt8](blobCipherText))
+        }
+
+        Function("expiringProfileKeyCredentialGetExpirationTime") { (sExpiringProfileKeyCredential: Data) -> Int64 in
+            let expkc = try ExpiringProfileKeyCredential(contents: [UInt8](sExpiringProfileKeyCredential))
+            return Int64(expkc.expirationTime.timeIntervalSince1970)
+        }
+
+        Function("profileKeyCredentialPresentationGetUuidCiphertext") { (sProfileKeyCredentialPresentation: Data) -> [UInt8] in
+            let pkcp = try ProfileKeyCredentialPresentation(contents: [UInt8](sProfileKeyCredentialPresentation))
+            return try pkcp.getUuidCiphertext().serialize()
+        }
+
+        Function("profileKeyCredentialPresentationGetProfileKeyCiphertext") { (sProfileKeyCredentialPresentation: Data) -> [UInt8] in
+            let pkcp = try ProfileKeyCredentialPresentation(contents: [UInt8](sProfileKeyCredentialPresentation))
+            return try pkcp.getProfileKeyCiphertext().serialize()
+        }
+
+        Function("profileKeyCredentialRequestContextGetRequest") { (sProfileKeyCredentialRequestContext: Data) -> [UInt8] in
+            let pkcrc = try ProfileKeyCredentialRequestContext(contents: [UInt8](sProfileKeyCredentialRequestContext))
+            return try pkcrc.getRequest().serialize()
+        }
+
+        Function("serverPublicParamsCreateProfileKeyCredentialRequestContextDeterministic") { (sServerPublicParams: Data, randomness: Data, fixedWidthAci: Data, sProfileKey: Data) -> [UInt8] in
+            let serverPublicParams = try ServerPublicParams(contents: [UInt8](sServerPublicParams))
+            let clientZkProfileOperation = ClientZkProfileOperations(serverPublicParams: serverPublicParams)
+            var bytes = convertDataToServiceIdStorage(data: fixedWidthAci)
+            let byteArray = try signalServiceIdServiceIdBinary(value: &bytes)
+            let uuid = UUID(uuid: (
+                byteArray[0], byteArray[1], byteArray[2], byteArray[3], 
+                byteArray[4], byteArray[5], byteArray[6], byteArray[7], 
+                byteArray[8], byteArray[9], byteArray[10], byteArray[11], 
+                byteArray[12], byteArray[13], byteArray[14], byteArray[15]
+            ))
+            let aci = Aci(fromUUID: uuid)
+            let profileKey = try ProfileKey(contents: [UInt8](sProfileKey))
+            guard randomness.count == 32 else {
+                throw NSError(domain: "Invalid input size", code: 1, userInfo: nil)
+            }
+            let randomnessBytes = randomness.withUnsafeBytes { pointer in
+                pointer.load(as: SignalRandomnessBytes.self)
+            }
+            return try clientZkProfileOperation.createProfileKeyCredentialRequestContext(randomness:  Randomness(randomnessBytes), userId: aci, profileKey: profileKey).serialize()
+        }
```

Comment:
fixed width binary bytes can be converted to into ServiceId through its constructor. and then raw uuid can be extracted with a simple method call.
libsignal/swift/Sources/LibSignalClient/ServiceId.swift:207